### ISSUE TITLE
Debugger disappearing/not active after a restart

### ISSUE THREAD

**Issue by balikian** [SqrTT/prophet#372]
In VS Code starting the debugger works until I hit the restart button.  Then it appears the debugger is turned off.  Breakpoints don't fire. The attached to sandbox bar is not seemingly on.  The play/restart buttons are gone.

Starting the debugger again will result in a warning: 'Attach to Sandbox' is already running.  Do you want to start another instance?

pressing [yes] turns everything back on again.  Hitting restart a second time will maintain the visual that the debugger is on, but it will not actually hit at any breakpoints.

Stopping the debugger requires you to stop both instances before you can start the debugger again correctly.

**To Reproduce**
1. in VS Code, set a breakpoint
2. start the debugger - F5 or Run - Start Debugging
3. notice the "Attach to sandbox" bar at the bottom indicates the debugger is on and the debugger controls at the top of the screen also show it's running.
4. Restart the debugger - either by hitting the restart button or ctrl+shift+F5 or choosing Run - Restart Debugging
5. notice both the bottom bar and the top controls indicate the debugger has stopped.  and in fact the debugger breakpoints do not hit.
6. Start the debugger again and notice the warning "'Attach to Sandbox' is already running.  Do you want to start another instance?"
7. choose yes and debugger is on and working again.
8. stopping the debugger requires you to stop it twice.

**Expected behavior**
Restarting the debugger should stop the instance and start it back up again.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 10
 - VSCode version 1.84.0
 - Version: 1.84.0 (user setup)
Commit: d037ac076cee195194f93ce6fe2bdfe2969cc82d
Date: 2023-11-01T11:29:04.398Z
Electron: 25.9.2
ElectronBuildId: 24603566
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Windows_NT x64 10.0.19045

 - The extension version 1.4.40

**Comment by SqrTT** [comment]
Hi @balikian 
I'm not sure if this is debugger issue itself. Since extension don't implement logic of run/stop debugger.
Does issue present before?

**Comment by balikian** [comment]
No.  I never had this problem until I did an update of VSCode maybe in September.  After that update, this issue started happening.  I'm re-training myself NOT to use the restart button.  If I let the debugger run through, it stays on and is fine.  If I stop it.  Then start it again manually, it works fine (outside of my debugger connecting to my build folder which requires many starts and stops).  If I restart it, this is the problem.
